### PR TITLE
TC03 - "User cannot log into Railway with invalid password "

### PR DESCRIPTION
**Pull request by viethuynh140499**

### CODE REVIEW

File: src/test/java/test_cases/LoginTests.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  public void TC01() {
+    User user = new User(Constant.USERNAME, Constant.PASSWORD);
+
+    homePage.clickLoginTab();
+    loginPage.login(user);
+
+    String actualResult = homePage.getWelcomeMessage();
+    String expectedResult = "Welcome " + user.getUsername();
+
+    Assert.assertEquals(actualResult, expectedResult);
```
Comment (fey031) [inline src/test/java/test_cases/LoginTests.java:25]: Add assert message.
Assert.assertEquals(actualResult, expectedResult, message);

File: src/test/java/test_cases/LoginTests.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  public void TC02() {
+    User user = new User("", Constant.PASSWORD);
+
+    homePage.clickLoginTab();
+    loginPage.login(user);
+
+    String actualResult = loginPage.getErrorMessage();
+    String expectedResult = "There was a problem with your login and/or errors exist in your form.";
+
+    Assert.assertEquals(actualResult, expectedResult);
```
Comment (fey031) [inline src/test/java/test_cases/LoginTests.java:38]: same above

File: src/test/java/test_cases/LoginTests.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    User user = new User(Constant.USERNAME, DataHelper.getRandomInvalidPassword());
+
+    homePage.clickLoginTab();
+    loginPage.login(user);
+
+    String actualResult = loginPage.getErrorMessage();
+    String expectedResult = "There was a problem with your login and/or errors exist in your form.";
+
+    Assert.assertEquals(actualResult, expectedResult);
```
Comment (fey031) [inline src/test/java/test_cases/LoginTests.java:52]: same above

Comment (viethuynh140499) [reply src/test/java/test_cases/LoginTests.java]: Yes, I added assert message

File: src/test/java/test_cases/LoginTests.java (hunk 2 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/LoginTests.java]: Yes, I added assert message

File: src/test/java/test_cases/LoginTests.java (hunk 1 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/LoginTests.java]: Yes, I added assert message

File: src/test/resources/testNG.xml [hunk 4]
```diff
@@ -3,20 +3,20 @@
...
         </classes>
     </test>
     <test name="EdgeTest">
         <parameter name="browser" value="EDGE" />
         <classes>
-            <class name="test_cases.TestCase01"/>
+            <class name="test_cases.LoginTests"/>
         </classes>
     </test>
 </suite>
```
Comment (tanmle) [inline src/test/resources/testNG.xml:22]: break line.

File: src/test/java/test_cases/LoginTests.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    String actualResult = loginPage.getErrorMessage();
+    String expectedResult = "There was a problem with your login and/or errors exist in your form.";
+
+    Assert.assertEquals(actualResult, expectedResult, "Error Message is not displayed");
+  }
+
+  @Test(description = "User cannot log into Railway with invalid password ")
+  public void TC03() {
+
```
Comment (tanmle) [inline src/test/java/test_cases/LoginTests.java:43]: redundant line

**Review by tanmle — CHANGES_REQUESTED** [review]

File: src/test/resources/testNG.xml (hunk 4 above)
Comment (viethuynh140499) [reply src/test/resources/testNG.xml]: yes, I added break line

File: src/test/java/test_cases/LoginTests.java (hunk 5 above)
Comment (viethuynh140499) [reply src/test/java/test_cases/LoginTests.java]: yes, I remove redundant line

File: src/test/java/test_cases/LoginTests.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    String actualResult = loginPage.getErrorMessage();
+    String expectedResult = "There was a problem with your login and/or errors exist in your form.";
+
+    Assert.assertEquals(actualResult, expectedResult, "Error Message is not displayed");
+  }
+
+  @Test(description = "User cannot log into Railway with invalid password ")
+  public void TC03() {
+    User user = new User(Constant.USERNAME, DataHelper.getRandomInvalidPassword());
```
Comment (fey031) [inline src/test/java/test_cases/LoginTests.java:43]: We don't need to use random password here, you can use any string for it

**Review by fey031 — APPROVED** [review]
LGTM

**Pull request merged**